src/main/java/frc/team166/robot/subsystems/Manipulator.java

```diff
@@ -0,0 +1,246 @@
+	}
+
+	/**
+	 * Resets encoders
+	 * <p>
+	 * Calls the reset function on the left and right intake encoders
+	 */
+	public void resetEncoders() {
+		leftIntakeEncoder.reset();
+		rightIntakeEncoder.reset();
+	}
+
+	/**
+	 * Gets IR voltage
+	 * <p>
+	 * Returns the voltage value output from the IR sensor
+	 * 
+	 * @return The voltage from the IR sensor
+	 */
+	public double getIRDistance() {
+		return irSensor.getVoltage(); //Manipulate this data pending experimentation
+	}
+
+	public void openManipulator() {
+		manipulatorSolenoid.set(Value.kForward);
+		;
+	}
+
+	public void closeManipulator() {
+		manipulatorSolenoid.set(Value.kReverse);
+	}
+
+	public double getAvgEncoderRate() { //ft/s
+		double leftRate = leftIntakeEncoder.getRate();
+		double rightRate = rightIntakeEncoder.getRate();
+		return (leftRate + rightRate) / 2.0;
+	}
+
+	/**
+	 * Gets average encoder distance
+	 * <p>
+	 * Returns the average distance of the left and right encoders
+	 * 
+	 * @return The average distance of the two encoders
+	 */
+	public double getAvgEncoderDistance() { //ft
+		double leftDist = leftIntakeEncoder.getDistance();
+		double rightDist = rightIntakeEncoder.getDistance();
+		return (leftDist + rightDist) / 2.0;
+	}
+
+	/**
+	 * Sets motors to intake mode
+	 * <p>
+	 * Sets the motors to 4/5 power inward to take in a cube on the field
+	 */
+	public void setMotorsToIntake() {
+		rollers.set(0.8); //change once you find optimal motor speed
+	}
+
+	/**
+	 * Sets motors to discharge mode
+	 * <p>
+	 * Sets the motors to 4/5 power outward to eject a stored cube
+	 */
+	public void setMotorsToDischarge() {
+		rollers.set(-0.8); //change once you find optimal motor speed
+	}
+
+	//COMMANDS
+	public void initDefaultCommand() {
+		setDefaultCommand(new SubsystemCommand(this) {
+
+			@Override
+			protected boolean isFinished() {
+				return false;
+			}
+
+		});
+	}
+
+	public Command DropCube() {
+		return new SubsystemCommand(this) {
+
+			@Override
+			protected void initialize() {
+				openManipulator();
+			}
+
+			@Override
+			protected void execute() {
+
+			}
+
+			@Override
+			protected boolean isFinished() {
+				return false;
+			}
+
+			@Override
+			protected void end() {
+
+			}
+		};
+	}
+
+	public Command CloseManipulator() {
+		return new SubsystemCommand(this) {
+
+			@Override
+			protected void initialize() {
+				closeManipulator();
+			}
+
+			@Override
+			protected void execute() {
+
+			}
+
+			@Override
+			protected boolean isFinished() {
+				return false;
+			}
+
+			@Override
+			protected void end() {
+
+			}
+		};
+	}
+
+	public Command CubePickup() {
+		return new SubsystemCommand(this) {
+
+			@Override
+			protected void initialize() {
+				resetEncoders();
+				rollers.set(0);
+				motorSpeed = 0;
+			}
+
+			@Override
+			protected void execute() {
+				//Figure out how to make the motors go 542.865 rpm or 6.81 ft/s without making them spool up
+				if (getAvgEncoderRate() < OPTIMAL_MOTOR_RATE)
```

Comment:
We should probably be using a PIDController instead of rolling our own.

That said, I think we should push back on the use of encoders here at all. I'm failing to see the benefit given the added complexity.